subsys/net/ip/connection.c

```diff
@@ -871,11 +873,17 @@ enum net_verdict net_conn_input(enum net_ip_protocol proto, struct net_pkt *pkt)
 #endif
 	{
 		send_icmp_error(pkt);
+
+		if (IS_ENABLED(CONFIG_NET_TCP) && proto == IPPROTO_TCP) {
+			net_stats_update_tcp_seg_connrst();
+		}
 	}
 
 drop:
 	if (proto == IPPROTO_UDP) {
 		net_stats_update_udp_drop();
+	} else if (IS_ENABLED(CONFIG_NET_TCP) && proto == IPPROTO_TCP) {
+		net_stats_update_tcp_seg_drop();
 	}
```

Comment:
Hum, why not wrapping this if/else if (as well as the update recv above) in an inline function such as: 
net_stats_update_per_proto_recv(proto);
and:
net_stats_update_per_proto_drop(proto);